### PR TITLE
SALTO-6428: Change Microsoft Security type names to PascalCase

### PR DESCRIPTION
**Pull request by shoshana-houri**
Change all Microsoft Security type names to PascalCase, following the discussion here

---
This was done following the discussion [here](https://github.com/salto-io/salto/pull/6456#discussion_r1727124764).

---
_Release Notes_: 

---
_User Notifications_:

### CODE REVIEW

File: packages/lowerdash/src/types.ts [hunk 1]
```diff
@@ -105,3 +105,31 @@ export type RecursivePartial<T> = {
       ? RecursivePartial<T[P]>
       : T[P]
 }
+
+export type UPPER_CASE_LETTERS =
```
Comment (shoshana-houri) [inline packages/lowerdash/src/types.ts:109]: i know it's kind of a lot 😅 but I think it's useful.
Let me know if you disagree or if you think we should locate this someplace else :)

**Review by shir-reifenberg — APPROVED** [review]
Added one comment, pre approving 👯

**Comment by shoshana-houri** [comment]
Thanks @shir-reifenberg! For some reason I can't see the comment  😅

File: packages/lowerdash/src/types.ts (hunk 1 above)
Comment (shir-reifenberg) [reply packages/lowerdash/src/types.ts:109]: IMO it's a bit of a overkill 😅 We can just make `toEntraTypeName`  to get a string

**Comment by shir-reifenberg** [comment]
> Thanks @shir-reifenberg! For some reason I can't see the comment 😅

Oops not sure why is wasn't submitted, added now 👀

File: packages/lowerdash/src/types.ts (hunk 1 above)
Comment (shoshana-houri) [reply packages/lowerdash/src/types.ts:109]: Oh well, it was worth the shot 😄

**Pull request merged**